src/Filesystem.cpp

```diff
@@ -33,48 +42,31 @@ enum MountPosition
 	MOUNT_APPEND = 1,
 };
 
-
-/**
- * Default c'tor.
- */
-Filesystem::Filesystem(): mVerbose(false)
-{}
-
-
 /**
  * Shuts down PhysFS and cleans up.
  */
 Filesystem::~Filesystem()
 {
-	if (PHYSFS_isInit()) { PHYSFS_deinit(); }
 	std::cout << "Filesystem Terminated." << std::endl;
 }
 
 
 /**
- * Shuts down PhysFS and cleans up.
+ * Initializes PhysFS
  */
-void Filesystem::init(const std::string& argv_0, const std::string& appName, const std::string& organizationName, const std::string& dataPath)
+void Filesystem::init(const std::string& /*argv_0*/, const std::string& appName, const std::string& organizationName, const std::string& dataPath)
 {
-	if (PHYSFS_isInit()) { throw filesystem_already_initialized(); }
+    namespace FS = std::filesystem;
+    if(mIsInit) { throw filesystem_already_initialized(); }
```

Comment:
Should be a call to function versus direct member access. This is a stylistic choice that I tend to use in all of my code, results in fewer changes should there be a different way to determine init status. Also, consistency.